`make manifests` fails

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [] I've pasted the output of `argocd version`. -> N/A

**Describe the bug**

* `make manifests` fails when I use WSL2 or Ubuntu 22.04 in virtualBox.

**To Reproduce**

1. `git clone https://github.com/argoproj/argo-cd/`
2. `cd argo-cd`
3. `make manifests`
4. Get a following error
```
+ helm dependency update ./chart
Getting updates for unmanaged Helm repositories...
...Unable to get an update from the "https://dandydeveloper.github.io/charts" chart repository:
        open /home/user/.cache/helm/repository/helm-manager-97a7e3aa4edbf42bb12d1f82bdf04f5b4e1d13793554e4ca19cfa6376b880787-index.yaml: no such file or directory
Error: no cached repository for helm-manager-97a7e3aa4edbf42bb12d1f82bdf04f5b4e1d13793554e4ca19cfa6376b880787 found. (try 'helm repo update'): open /home/user/.cache/helm/repository/helm-manager-97a7e3aa4edbf42bb12d1f82bdf04f5b4e1d13793554e4ca19cfa6376b880787-index.yaml: no such file or directory
make: *** [Makefile:260: manifests-local] Error 1
make: *** [Makefile:264: manifests] Error 2
```

**Expected behavior**

* Successfully generate manifests.

**Screenshots**

* N/A

**Version**

* N/A

**Logs**

* N/A

**Considerations**

* The root cause may be directory permission of `/home/user/.cache`.
  * `docker run --rm -it --name argocd-test-client -u 1000:1000 -e HOME=/home/user -e GOPATH=/go -e ARGOCD_E2E_K3S= -e GITHUB_TOKEN -e GOCACHE=/tmp/go-build-cache -e ARGOCD_LINT_GOGC=20 argocd-test-tools:latest bash`
  * `ls -la ~/`
   ```
   drwxr-xr-x 4 user user 4096 Feb 20 02:49 .
   drwxr-xr-x 1 root root 4096 Feb 20 02:49 ..
   drwxr-xr-x 2 root root 4096 Feb 20 02:49 .cache
   -rw-r--r-- 1 root root   61 Feb 20 02:49 .gitconfig
   drwxr-xr-x 2 user user 4096 Feb 20 02:49 .kube
   ```
* https://github.com/argoproj/argo-cd/blob/ba86be2fcbab6c53c311426a31d9761ff6934300/test/container/Dockerfile#L109

--- a/test/container/Dockerfile
+++ b/test/container/Dockerfile
@@ -95,6 +95,7 @@ ARG UID
 RUN useradd -l -u ${UID} -d /home/user -s /bin/bash user && \
     echo "user ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/user && \
     mkdir -p /home/user/.kube && \
+    mkdir -p /home/user/.cache && \
     chown -R user /home/user && \
     chgrp -R user /home/user && \
     HOME=/home/user git config --global user.name "ArgoCD Test User" && \
@@ -106,7 +107,6 @@ RUN useradd -l -u ${UID} -d /home/user -s /bin/bash user && \
     chown root /etc/ssh/ssh_host_*_key* && \
     chmod 0600 /etc/ssh/ssh_host_*_key && \
     mkdir -p /tmp/go-build-cache && \
-    mkdir -p /home/user/.cache && \
     ln -s /usr/local/bin/node /usr/local/bin/nodejs && \
     ln -s /usr/local/lib/node_modules/npm/bin/npm-cli.js /usr/local/bin/npm && \
     ln -s /usr/local/lib/node_modules/npm/bin/npx-cli.js /usr/local/bin/npx && \

